source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -545,8 +569,26 @@ void Filter::setEncoderFilterCallbacks(Http::StreamEncoderFilterCallbacks& callb
   watermark_callbacks_.setEncoderFilterCallbacks(&callbacks);
 }
 
-void Filter::sendRequest(ProcessingRequest&& req, bool end_stream) {
+void Filter::sendRequest(const ProcessorState& state, ProcessingRequest&& req, bool end_stream) {
   // Calling the client send function to send the request.
+  if (!processing_request_modifier_initialized_) {
```

Comment:
Recommend adding this code block to create the processing_request_modifier_ be done in the configuration parsing phase https://github.com/envoyproxy/envoy/blob/317d7fdf8cac9f4e7f677540e68d16c7f9de72c5/source/extensions/filters/http/ext_proc/config.cc#L76?  .  Also, if you do it there, there is no need for the new boolean flag processing_request_modifier_initialized_.